calc.py

```diff
@@ -0,0 +1,56 @@
+import math
+
+
+def add(a, b):
+    check_types(a, b)
+    return a + b
+
+
+def subtract(a, b):
+    check_types(a, b)
+    return a - b
+
+
+def multiply(a, b):
+    check_types(a, b)
+    return a * b
+
+
+def log(a):
+    check_type(a)
+    return math.log(a)
+
+
+def log10(a):
+     if a > 0:
+      check_type(a)
```

Comment:
check_type(a) call should be made before checking whether a > 0. What if a is None, in which case 'if a > 0' is going to fail (since Python treats None as equal to 0) and the customers will get an incorrect error message.